### PR TITLE
switching token endpoint off of matrix

### PR DESCRIPTION
**Pull request by genslein**
Since we are moving the token endpoint off of the matrix, the sdk needs to be updated to show the new location. Please double check and make sure I did not miss anything in the sdk.

### CODE REVIEW

**Comment by sakai135** [comment]
:+1:

**Comment by genslein** [comment]
Just to confirm, I didn't see anything on the sdk that sets www.careerbuilder.com or www.careerbuilder.eu, those will need to be switched to api.careerbuilder.com and .eu respectively

**Comment by sakai135** [comment]
Thanks, we currently override the default in here by passing in the appropriate `base_url` value through the constructor. @genslein what's the timeline for the switchover?

**Comment by genslein** [comment]
Dual support is currently active, though we are actively asking people to begin switching. We would like to sunset token.aspx in the matrix within the next month or so. I'd have to ask @ahodges for exact timeline as I'm not sure.

**Comment by christian-blades-cb** [comment]
Hey @jpalmour - We'll need to change the [URLs]https://github.com/cbdr/luceo/blob/master/app/config/platform.yml#L55) for each region, and look into how we do the same for <4.0 clients.

**Comment by jpalmour** [comment]
Ok, I suppose this shouldn't be merged until I've updated platform.yml, then. We'll need some type of coordination, at least. When will this be ready to go live? Monday night's scheduled deployment?

I'm not sure, but I imagine I'll need to update config_global.php to get this working for pre 4.0. 

Although if we used a new platform.yml config_global.php key, I guess coordination wouldn't be nearly as tricky, since we could have the old and new in place simultaneously.

**Comment by sakai135** [comment]
@genslein correct me if I'm wrong, but `/share/oauth2/token.aspx` works with both the old and new hosts, right? If so, then we can just update the hosts, then update this OAuth plugin version after that.

**Comment by christian-blades-cb** [comment]
@KaiTillman @dsmith-cb @rjashman - Does this URL need to change as well? https://github.com/cbdr/luceo/blob/master/app/config/platform.yml#L61

**Comment by genslein** [comment]
@sakai135 @christian-blades-cb @jpalmour ```www.careerbuilder.[region]/share/oauth2/token.aspx```` is the matrix endpoint. Since we are trying to improve performance for customers by going directly to the actual service off matrix, those go through ```api.careerbuilder.[region]/oauth/token```. Both are active right now so you can test side by side.

@sakai135 to answer your question, that endpoint is public as is the new endpoint, though the idea is to get rid of the matrix endpoint ```share/oauth2/token.aspx``` in favor of latency performance gains.

**Comment by sakai135** [comment]
Ok, so `https://api.careerbuilder.com/share/oauth2/token.aspx` still works.

**Comment by genslein** [comment]
@sakai135 Keichii, it should always have been ```https://www.careerbuilder.com/share/oauth2/token.aspx``` this PR changes to use ```api.careerbuilder.com``` and the endpoint is ```/oauth/token```

**Comment by genslein** [comment]
@sakai135 as an update, I see a TalentStream Recruit App calling ```wwwtest.api.careerbuilder.com/share/oauth2/token.aspx```. This is adding double the latency because this hits our Api routing layer, back to matrix, then back to routing, routes to the Authorization endpoint, then all the way back zig zagging until it returns.

tl;dr version: you average 350 ms, changing to the proper call averages 91 ms.

**Comment by sakai135** [comment]
@genslein Right, it's just that using `https://api.careerbuilder.com/share/oauth2/token.aspx` would allow us to switch configurations without worrying about syncing the configuration change and the version change. Once it's done, we'll be on the proper endpoint.
